src/Core/Commander/Providers/WMS_Provider.js

```diff
@@ -95,125 +81,57 @@ WMS_Provider.prototype.getColorTexture = function getColorTexture(tile, layer, b
-        result.texture.needsUpdate = true;
+    return this.getColorTextureByUrl(url).then((texture) => {
+        result.texture = texture;
+        result.texture.coordWMTS = tile.wmtsCoords[layer.options.tileMatrixSet][0];
+        result.texture.bbox = bbox;
         return result;
     });
 };
 
 WMS_Provider.prototype.getXbilTexture = function getXbilTexture(tile, layer, bbox, pitch) {
-    var url = this.url(bbox, layer);
-
-    // TODO: this is not optimal: if called again before the IoDriver resolves, it'll load the XBIL again
-    var textureCache = this.cache.getRessource(url);
-
-    if (textureCache !== undefined) {
-        return Promise.resolve({
-            pitch,
-            texture: textureCache.texture,
-            min: textureCache.min,
-            max: textureCache.max,
-        });
-    }
-
-    // bug #74
-    // var limits = layer.tileMatrixSetLimits[coWMTS.zoom];
-    // if (!limits || !coWMTS.isInside(limits)) {
-    //     var texture = -1;
-    //     this.cache.addRessource(url, texture);
-    //     return Promise.resolve(texture);
-    // }
-    // -> bug #74
-    return this._IoDriver.read(url).then((result) => {
-        result.texture = this.getTextureFloat(result.floatArray);
-        result.texture.generateMipmaps = false;
-        result.texture.magFilter = THREE.LinearFilter;
-        result.texture.minFilter = THREE.LinearFilter;
-        result.pitch = pitch;
-
-        // In RGBA elevation texture LinearFilter give some errors with nodata value.
-        // need to rewrite sample function in shader
-        this.cache.addRessource(url, result);
-
+    const url = this.url(bbox, layer);
+    return this.getXBilTextureByUrl(url, pitch).then((result) => {
+        result.texture.coordWMTS = tile.wmtsCoords[layer.options.tileMatrixSet][0];
         return result;
     });
 };
 
 WMS_Provider.prototype.executeCommand = function executeCommand(command) {
-    var layer = command.layer;
-    var tile = command.requester;
-    var ancestor = command.ancestor;
+    const layer = command.layer;
+    const tile = command.requester;
 
-    var supportedFormats = {
+    computeTileWMTSCoordinates(tile, layer, this.projection);
+
+    const supportedFormats = {
         'image/png': this.getColorTexture.bind(this),
         'image/jpg': this.getColorTexture.bind(this),
         'image/jpeg': this.getColorTexture.bind(this),
         'image/x-bil;bits=32': this.getXbilTexture.bind(this),
     };
 
-    var func = supportedFormats[layer.format];
+    const func = supportedFormats[layer.format];
     if (func) {
-        var pitch = ancestor ?
-            this.projection.WMS_WGS84Parent(tile.bbox, ancestor.bbox) :
-            new THREE.Vector3(0, 0, 1);
-
-        var bbox = ancestor ?
-            ancestor.bbox :
-            tile.bbox;
-
+        const searchInParent = tile.materials[0].getColorLayerLevelById(layer.id) < 0 && tile.parent.materials[0].getColorLayerLevelById(layer.id) > -1;
```

Comment:
No, you can't depend on `tile.materials[0].getColorLayerLevelById(layer.id)` to do this.
Actually you can't depend on `LayeredMaterial` at all.
Add the needed params to the `command` instead (for instance the parent's texture for this layer).